proposals/fetching.md

```diff
@@ -23,7 +30,23 @@ Currently, platforms and templates are cached at `~/.cordova/lib/npm_cache/PACKA
 
 ### Proposal:
 
-* Create a new module named `cordova-fetch`. Fetches from npm, git and local paths. 
+* Create a new module named `cordova-fetch`. Fetches from npm and git. 
 * Start using system installed `npm` instead of packaging our own `npm` in `cordoba-lib`. We already check for `git` being installed, we should do the same for `npm`. We would use  `superspawn` to shell out to the system `npm`.
-* Stop using `.cordova/npm_cache/` and instead use `.npm/` for all of our `npm caching`. This will allow us to move away from passing custom config options to npm. 
-* Update existing fetching logic to use new `cordova-fetch` module.
+* Stop using `.cordova/npm_cache/`. Modules instead get fetched to root `node_modules` directory in your applications
+* Add new `--fetch` flag to `cordova` and `plugman` to use `cordova-fetch` over existing fetching methods
+* Plugin/platform removal should run `npm uninstall` on your cordova projects to remove the module from the apps `node_modules` directory.
+* if a `package.json` exists, adding `--save` will also add the dependency to `package.json` (as well as to `config.xml`). This won't be used until a future update is made to move save functionality from `config.xml` to `package.json`. Just adding it now for future proofing.
+
+### New Requirements
+
+* Have to remove support for installing plugins/platforms from git subdirectoires. https://cordova.apache.org/docs/en/5.0.0/guide/cli/#advanced-plugin-options. I recommend we add deprecation notices for this ASAP.
+* Every plugin, platform and template requires a `package.json`
+
+### Quirks
+
+* Currently, I am using a diff of `npm ls` to determine which module was just installed. This is because the user can pass in a git url as a target to `cordova-fetch`. If the git url repo-name is different than the `package.name`, only way to get what module was just installed is by doing a diff before and after the installation. The diffing technique can fail if the user is adding a module that has already been npm installed. Luckily cordova doesn't allow adding platform/plugin which has already been added. cordova platform/plugin rm will also remove the module from the applications `node_modules` directory.
```

Comment:
cordova prepare currently restores based on what is saved in config.xml and what is installed. I would think it would continue to do the restore step but use package.json instead of config.xml. 

So if a user runs `npm install`, then `cordova prepare`, it would see that those plugins & platforms aren't "installed" according to cordova, so it would go through the motions of fetching (Which is already done since the user ran `npm install`), then copying to platforms/plugins. 

If a user doesn't run `npm install`, cordova prepare would detect that those plugins/platforms aren't "installed" and move forward with fetching. 

So cordova can support both usecases 😄